workspace_host/interface.js

```diff
@@ -776,19 +788,33 @@ function _recursiveDownloadJobManager(curDirPath, S3curDirPath, callback) {
 async function _getInitialZipAsync(workspace) {
     workspaceHelper.updateMessage(workspace.id, 'Loading initial files');
     const localName = workspace.local_name;
-    const s3Name = workspace.s3_name;
-    const localPath = `${workspacePrefix}/${localName}`;
+    const remoteName = workspace.remote_name;
+    let localPath;
     const zipPath = `${config.workspaceHostZipsDirectory}/${localName}-initial.zip`;
-    const s3Path = `${s3Name}/initial.zip`;
+    const remotePath = `${remoteName}/initial.zip`;
 
-    debug(`Downloading s3Path=${s3Path} to zipPath=${zipPath}`);
-    const options = {
-        owner: config.workspaceJobsDirectoryOwnerUid,
-        group: config.workspaceJobsDirectoryOwnerGid,
-    };
-    const isDirectory = false;
-    update_queue[[zipPath, isDirectory]] = { action: 'skip' };
-    await awsHelper.downloadFromS3Async(config.workspaceS3Bucket, s3Path, zipPath, options);
+    debug(`Downloading remotePath=${remotePath} to zipPath=${zipPath}`);
+    if (workspace.homedir_location == 'S3') {
+        localPath = `${workspacePrefix}/${localName}`;
+
+        const isDirectory = false;
+        update_queue[[zipPath, isDirectory]] = { action: 'skip' };
+        const options = {
+            owner: config.workspaceJobsDirectoryOwnerUid,
+            group: config.workspaceJobsDirectoryOwnerGid,
+        };
+        await awsHelper.downloadFromS3Async(config.workspaceS3Bucket, remotePath, zipPath, options);
+    } else if (workspace.homedir_location == 'FileSystem') {
```

Comment:
I'm a bit confused here. If we have a workspace with FileSystem storage, why do we need to do any copying at all? Won't the files automatically be in the right place when we start up?

Reply:
I'm also confused 🙂 

I'm not exactly sure when the `initial.zip` file is used (if at all), so I left this in there.  If it turns out that we don't actually need it I'm happy to remove this bit.